### PR TITLE
Update gardener/gardener to v1.86.0

### LINKED ISSUE #734: [Feature] Bump Gardener dependency
**Feature (What you would like to be added):**

Currently the `etcd-druid` project is using a pretty old Gardener dependency (`v1.76.0`). Is that intended or can the Gardener version be bumped to a post 1.8x version?

### PR DESCRIPTION
**Pull request by shreyas-s-rao**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:
This PR updates the gardener/gardener dependency to v1.86.0, bringing in multiple other changes:
- [x] Remove vendor directory, in line with gardener, similar to https://github.com/gardener/gardener/pull/8775
- [x] Bump dependencies, required for g/g v1.86.0
- [x] Update paths in Makefile, CI scripts and hack scripts to remove usage of vendor dir for g/g hack scripts, and find the g/g dir in mod cache instead
- [x] Remove `-mod=vendor` from builds, runs and test runs
- [x] Run `make tidy`, which pulls in new CI hack scripts and issue/PR templates
- [x] Adapt and run `make generate`
- [x] Remove unused chart renderer code from g/g, which was deprecated and now removed in the latest g/g version
- [x] Adapt manager and controller registration code to new controller-runtime version, both in main code and integration test utils
- [x] Replace `github.com/golang/mock/gomock` with `go.uber.org/mock/gomock`, as stated in https://github.com/golang/mock 
- [x] Adapt etcdcopybackupstask deletion unit tests to new controller-runtime fake client behavior. There is a change in behavior of [fake client handling of deletion timestamps](https://github.com/kubernetes-sigs/controller-runtime/pull/2316), as well as [handling updates to status subresource of CRDs](https://github.com/kubernetes-sigs/controller-runtime/pull/2259), due to which the tests needed to be adapted 
- [x] Downgrade kustomize from v5.0.0 to v4.5.7 due to [this bug](https://github.com/kubernetes-sigs/kustomize/issues/4731), which we and many other users used as a feature. #765 details out the issue and a recommendation to properly fix this in the long-term, but this PR should go ahead with the checked in fix.
- [x] Update github issue templates with new ones from g/g.

**Which issue(s) this PR fixes**:
Fixes #734 

**Special notes for your reviewer**:
/cc @afritzler @shafeeqes 
/invite @seshachalam-yv @ishan16696 @unmarshall 

It's easier to review this PR commit by commit, rather than all files as once, for two reasons: the commits are well-structured and segregated by individual purposes, and the removal of vendor dir causes a large diff, making it difficult for github to render the diff page. You may also review the [set of commits](https://github.com/gardener/etcd-druid/pull/748/files/601216f72233e81f37e3682224f6b9d4d24ca6ec..01ade834e7f6b088eaf7be75405ce544d4917b44#diff-d900babf741981553f1ae363bc2d0062aab6a95256dfa091b688df589e9a5c5d) omitting the first commit which removes the vendor dir.

**Release note**:

```breaking developer
Vendor directory has now been removed from the project. Please run `make tidy` to pull dependencies into go mod cache initially, and whenever required.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,45 +12,35 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Image URL to use all building/pushing image targets
+ENSURE_GARDENER_MOD := $(shell go get github.com/gardener/gardener@$$(go list -m -f "{{.Version}}" github.com/gardener/gardener))
+GARDENER_HACK_DIR   := $(shell go list -m -f "{{.Dir}}" github.com/gardener/gardener)/hack
 VERSION             := $(shell cat VERSION)
 REPO_ROOT           := $(shell dirname "$(realpath $(lastword $(MAKEFILE_LIST)))")
+HACK_DIR 			:= $(REPO_ROOT)/hack
```
Comment (shafeeqes) [inline Makefile:19]: Indentation

File: controllers/etcdcopybackupstask/reconciler_test.go [hunk 2]
```diff
@@ -84,36 +84,54 @@ var _ = Describe("EtcdCopyBackupsTaskController", func() {
...
 
-			It("should remove finalizer for task which does not have a corresponding job", func() {
+			It("should remove finalizer and delete task if it does not have a corresponding job", func() {
 				Expect(controllerutils.AddFinalizers(ctx, fakeClient, task, common.FinalizerName)).To(Succeed())
-				Expect(addDeletionTimestampToTask(ctx, task, time.Now(), fakeClient)).To(Succeed())
+				// use fakeClient.Delete() to simply add deletionTimestamp to `task` object,
+				// due to https://github.com/kubernetes-sigs/controller-runtime/pull/2316
+				Expect(fakeClient.Delete(ctx, task)).To(Succeed())
+				// get the updated object after deletionTimestamp has been added by fakeClient.Delete() call
+				Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(task), task)).To(Succeed())
```
Comment (shafeeqes) [inline controllers/etcdcopybackupstask/reconciler_test.go:134]: Doesn't the `Delete` call already work on a pointer to the object? Is this `Get` call still needed?

**Review by shafeeqes — COMMENTED** [review]
Thanks for the well-structured PR. Only minor comments from my side.

File: controllers/etcdcopybackupstask/reconciler_test.go (hunk 2 above)
Comment (shreyas-s-rao) [reply controllers/etcdcopybackupstask/reconciler_test.go:134]: > Doesn't the Delete call already work on a pointer to the object?

Yes, but within the deletion logic of fake client, it [adds deletion timestamp on a copy of the object](https://github.com/kubernetes-sigs/controller-runtime/blob/5f8d96b043ef8ccadc544535415395be336630e8/pkg/client/fake/client.go#L1031-L1047) (yes, quite weird). So the original object we had passed then differs from the actual object in the cluster, so we are hit with this error:

```
 [FAILED] Expected
      <*fmt.wrapError | 0x14000781aa0>: 
      could not remove finalizer: Operation cannot be fulfilled on etcdcopybackupstasks.druid.gardener.cloud "test-etcd-copy-backups-task": object was modified
      {
          msg: "could not remove finalizer: Operation cannot be fulfilled on etcdcopybackupstasks.druid.gardener.cloud \"test-etcd-copy-backups-task\": object was modified",
          err: <*errors.StatusError | 0x140005ff720>{
              ErrStatus: {
                  TypeMeta: {Kind: "", APIVersion: ""},
                  ListMeta: {
                      SelfLink: "",
                      ResourceVersion: "",
                      Continue: "",
                      RemainingItemCount: nil,
                  },
                  Status: "Failure",
                  Message: "Operation cannot be fulfilled on etcdcopybackupstasks.druid.gardener.cloud \"test-etcd-copy-backups-task\": object was modified",
                  Reason: "Conflict",
                  Details: {
                      Name: "test-etcd-copy-backups-task",
                      Group: "druid.gardener.cloud",
                      Kind: "etcdcopybackupstasks",
                      UID: "",
                      Causes: nil,
                      RetryAfterSeconds: 0,
                  },
                  Code: 409,
              },
          },
      }
  to be nil
  In [It] at: /Users/i349079/go/src/github.com/gardener/etcd-druid/controllers/etcdcopybackupstask/reconciler_test.go:137 @ 01/09/24 12:20:24.181
```

The only way to circumvent this error is to get the object again and then continue.

**Comment by shreyas-s-rao** [comment]
@shafeeqes thanks for your review. I have addressed both comments. PTAL

File: controllers/etcdcopybackupstask/reconciler_test.go (hunk 2 above)
Comment (shafeeqes) [reply controllers/etcdcopybackupstask/reconciler_test.go:134]: Weird, but anyways, for the tests, we can live with it.

**Comment by shreyas-s-rao** [comment]
Something seems wrong with the prow jobs. e2e tests are passing when run locally:
[druid-e2e.log](https://github.com/gardener/etcd-druid/files/13948302/druid-e2e.log)

**Comment by shreyas-s-rao** [comment]
@ishan16696 @shafeeqes I have now fixed the prow tests via https://github.com/gardener/etcd-druid/pull/748/commits/01ade834e7f6b088eaf7be75405ce544d4917b44, and also added an issue for a long-term fix: #765 .

PTAL and LGTM, thanks.

**Review by shafeeqes — COMMENTED** [review]
/lgtm

**Review by unmarshall — APPROVED** [review]

**Comment by unmarshall** [comment]
I have successfully run the following make targets:

1. `make check`
2. `make test`
3. `make test-integration`

The e2e are currently failing locally.
`make ci-e2e-kind`

**Pull request merged**